test/integration/integration_test.go

```diff
@@ -279,31 +288,57 @@ func checkConfigKey(configPath string, condition string, keyPath string) error {
 
 func compareExpectedWithActualContains(expected string, actual string) error {
 	if !strings.Contains(actual, expected) {
-		return fmt.Errorf("Output did not match. Expected: %s, Actual: %s", expected, actual)
+		return fmt.Errorf("Output did not match. Expected: '%s', Actual: '%s'", expected, actual)
 	}
 
 	return nil
 }
 
 func compareExpectedWithActualNotContains(notexpected string, actual string) error {
 	if strings.Contains(actual, notexpected) {
-		return fmt.Errorf("Output did match. Not expected: %s, Actual: %s", notexpected, actual)
+		return fmt.Errorf("Output did match. Not expected: '%s', Actual: '%s'", notexpected, actual)
 	}
 
 	return nil
 }
 
 func compareExpectedWithActualEquals(expected string, actual string) error {
 	if actual != expected {
-		return fmt.Errorf("Output did not match. Expected: %s, Actual: %s", expected, actual)
+		return fmt.Errorf("Output did not match. Expected: '%s', Actual: '%s'", expected, actual)
 	}
 
 	return nil
 }
 
 func compareExpectedWithActualNotEquals(notexpected string, actual string) error {
 	if actual == notexpected {
-		return fmt.Errorf("Output did match. Not expected: %s, Actual: %s", notexpected, actual)
+		return fmt.Errorf("Output did match. Not expected: '%s', Actual: '%s'", notexpected, actual)
+	}
+
+	return nil
+}
+
+func compareExpectedWithActualMatches(expected string, actual string) error {
+	expectedRegexp, err := regexp.Compile(expected)
+	if err != nil {
+		return fmt.Errorf("Expected value must be a valid regular expression statement: ", err)
+	}
+
+	if !expectedRegexp.MatchString(actual) {
+		return fmt.Errorf("Output did not match. Expected: '%s', Actual: '%s'", expected, actual)
+	}
+
+	return nil
+}
+
+func compareExpectedWithActualNotMatches(notexpected string, actual string) error {
```

Comment:
Duplication of code... the difference between `!expectedRegexp` and `notexpectedRegexp` is so minimal, so please create a function that can be reused:

  * it compiles the regex
  * it compares

In that case `compareExpectedWithActualMatches` and `compareExpectedWithActualNotMatches` re-use functionality and become much shorter.